cirq-core/cirq/transformers/randomized_measurements.py

```diff
@@ -13,97 +13,158 @@
 # limitations under the License.
 
 from collections.abc import Sequence
-from typing import Any, Literal
+from typing import Any
 
 import cirq
 import numpy as np
+from cirq.ops import SingleQubitCliffordGate
 from cirq.transformers import transformer_api
 
 
 @transformer_api.transformer
 class RandomizedMeasurements:
-    """A transformer that appends a moment of random rotations to map qubits to
-    random pauli bases."""
+    """A transformer that appends a moment of random rotations from a given unitary ensemble (pauli,
+    clifford, cue)"""
 
     def __init__(self, subsystem: Sequence[int] | None = None):
         """Class structure for performing and analyzing a general randomized measurement protocol.
         For more details on the randomized measurement toolbox see https://arxiv.org/abs/2203.11374
 
         Args:
             subsystem: The specific subsystem (e.g qubit index) to measure in random basis
+            rest of the qubits are measured in the computational basis
         """
         self.subsystem = subsystem
 
     def __call__(
         self,
-        circuit: 'cirq.AbstractCircuit',
+        circuit: "cirq.AbstractCircuit",
+        unitary_ensemble: str = "pauli",
         rng: np.random.Generator | None = None,
         *,
         context: transformer_api.TransformerContext | None = None,
-    ):
+    ) -> "cirq.Circuit":
         """Apply the transformer to the given circuit. Given an input circuit returns
-        a list of circuits with the pre-measurement unitaries. If no arguments are specified,
-        it will default to computing the entropy of the entire circuit.
+        a new circuit with the pre-measurement unitaries and measurements gates added.
+        to the qubits in the subsystem provided.If no subsystem is specified in the
+        construction of this classit defaults to measuring all the qubits in the
+        randomized bases.
 
         Args:
             circuit: The circuit to add randomized measurements to.
-            rng: Random number generator.
+            unitary_ensemble: Choice of unitary ensemble (pauli/clifford/cue(circular unitary ensemble))
             context: Not used; to satisfy transformer API.
-
+            rng: Random number generator.
+            
         Returns:
-            List of circuits with pre-measurement unitaries and measurements added
+            A circuit with pre-measurement unitaries and measurements added
         """
+
```

Comment:
nit: consider adding `unitary_ensemble = lower(unitary_ensemble)` to add support for users passing "Pauli" etc...

Reply:
done